pkg/handlers/util.go

```diff
@@ -51,3 +55,41 @@ func waitForPodSuccess(podClient resource.Client, podName string, podNS string,
 func waitPoll(waitFunc func() (bool, error), timeout time.Duration) error {
 	return wait.Poll(1*time.Second, timeout, waitFunc)
 }
+
+// Returns a strategic patch for adding or removing a label for a node. Operation can be add or delete.
+func patchForNodeWithLabel(oldNode *corev1.Node, label string, operation string, nodeClient resource.Client) (err error) {
```

Comment:
@Ajay191191 This seems pretty complicated. Can we get away with doing something dumb like https://github.com/kubernetes-incubator/node-feature-discovery/blob/master/main.go#L249-L266?